azurerm/resource_arm_virtual_machine_managed_disks_test.go

```diff
@@ -225,6 +225,33 @@ func TestAccAzureRMVirtualMachine_bugAzureRM33(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMVirtualMachine_changeStorageDataDiskCreationOption(t *testing.T) {
+	var afterCreate, afterUpdate compute.VirtualMachine
+	ri := acctest.RandInt()
+	preConfig := testAccAzureRMVirtualMachine_basicLinuxMachine_managedDisk_empty(ri, testLocation())
+	postConfig := testAccAzureRMVirtualMachine_basicLinuxMachine_managedDisk_attach(ri, testLocation())
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMVirtualMachineDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: preConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualMachineExists("azurerm_virtual_machine.test", &afterCreate),
+				),
+			},
+			{
+				Config: postConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualMachineExists("azurerm_virtual_machine.test", &afterUpdate),
+					testAccCheckVirtualMachineRecreated(t, &afterCreate, &afterUpdate),
```

Comment:
I think we should add a state check for the field that's being patched in this PR, just for sanity...